### PR TITLE
force delete all works in hosted mode

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: Wei Liu <liuweixa@redhat.com>

### CODE REVIEW

**Comment by skeeey** [comment]
refer to https://github.com/stolostron/backlog/issues/20960

File: pkg/controller/hosted/controller.go [hunk 1]
```diff
@@ -182,24 +187,57 @@ func klusterletNamespace(managedCluster string) string {
...
+// getHostedManifestWorks gets klusterlet and managed kubeconfig manifest works in the management cluster namespace
+func (r *ReconcileHosted) getHostedManifestWorks(ctx context.Context, cluster *clusterv1.ManagedCluster) ([]workv1.ManifestWork, error) {
+	hostedManifestWorks := []workv1.ManifestWork{}
+
+	managementCluster, err := helpers.GetHostingCluster(cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	klusterletManifestWork := &workv1.ManifestWork{}
```
Comment (zhujian7) [inline pkg/controller/hosted/controller.go:199]: The code of getting the klusterlet manifestwork and getting the kubeconfig manifestwork is a little bit repetitive, could you extract the logic to a function?

Comment (skeeey) [reply pkg/controller/hosted/controller.go]: updated

**Pull request merged**